languages/javascript/exercises/concept/closures/.docs/after.md

```diff
@@ -0,0 +1,111 @@
+The name **closure** is historically derived from [_λ-calculus_][wiki-lambda-calculus] and popularized by [_scheme_][wiki-scheme] ([source][wiki-closure]) to reference a function's open and closed variable bindings. **Closures** as they have been explored in this exercise are closely related to variable scope. In JavaScript, there are 3 types of variable scope:
+
+1. **Global-scope**: These are variables that are declared outside of functions and blocks.
+1. **Function-scope**: These are variables declared within a function using the `var` keyword.
+1. **Block-scope**: These are variables declared within a block (`{...}`) using the `let` and `const` keywords.
+
+In practice, a closure often occurs when a function (or block) uses a variable from an outer scope. In JavaScript this is supported transparently, so closures are often used without specific reference.
+
+## Reasons to use closures
```

Comment:
Did you consider moving this section _after_ the `## Lexical Scope for variables` section? To me that feels a bit more natural, with the 3 types of variables scopes just having been discussed in the above section and the fact that partial application is a somewhat more complicated concept.

Reply:
I had, but I think that it always helps me to know _the reasons why_ I should learn something before I _learn_ it, so I had originally organised it that way.  I can understand that this feels like a bit of a yo-yo with the back and forth, so I'll see what I can do.

Reply:
See new commit